src/api/app/models/status/report.rb

```diff
@@ -1,36 +1,37 @@
-class Status::RepositoryPublish < ApplicationRecord
+class Status::Report < ApplicationRecord
+  self.table_name = 'status_reports'
+
   #### Includes and extends
 
   #### Constants
 
   #### Self config
 
   #### Attributes
-  validates :repository, :build_id, presence: true
 
   #### Associations macros (Belongs to, Has one, Has many)
-  belongs_to :repository
-  has_many :checks, as: :checkable, dependent: :destroy
-  has_one :project, through: :repository
-  has_many :relationships, through: :project
-  has_many :groups, through: :relationships
-  has_many :groups_users, through: :groups
+  belongs_to :checkable, polymorphic: true
+  has_many :checks, class_name: 'Status::Check', dependent: :destroy, foreign_key: 'status_reports_id'
 
   #### Callbacks macros: before_save, after_save, etc.
 
   #### Scopes (first the default_scope macro if is used)
 
   #### Validations macros
+  validates :checkable, presence: true
+  validates :uuid, presence: true, if: proc { |record| record.checkable.is_a?(Repository) }
+  validates :uuid_type, presence: true, if: proc { |record| record.checkable.is_a?(Repository) }
```

Comment:
Why do we need a uuid_type at the moment?

Reply:
That's what we agreed on, the three of us. Right now it's just a bucket where you can write in something. We probably automatically derive this from the checkable type.

But I wouldn't worry about this to much right now. Since we are still developing the whole feature.

Reply:
I remember that we talked about it. However, as we don't use it I would drop it for now. The type can be derived from the checkable. This only gets relevant if you have two different types per checkable which we don't have a use case for it (and I doubt we will ever have).